src/time_travel/patchers/select_patcher.py

```diff
@@ -0,0 +1,64 @@
+"""A patch to the datetime module."""
+
+from basic_patcher import BasicPatcher
+
+import select as select_lib
+import mock
+
+
+class SelectPatcher(BasicPatcher):
+    """Patcher for select module.
+    
+    patching:
+        - select.select
+    """
+    
+    def __init__(self, *args, **kwargs):
+        """Create the patch."""
+        super(SelectPatcher, self).__init__(*args, **kwargs)
+        
+        self.select = mock.Mock(
+            side_effect=self._mocked_select,
+            spec=select_lib.select)
+        
+        self.patches = [mock.patch('select.select', self.select)]
+        
+    @staticmethod
+    def _list_intersection(list1, list2):
+        return list(set(list1).intersection(set(list2)))
+        
+    def _mocked_select(self, rlist, wlist, xlist, timeout=None):
+        waited_events = set(rlist + wlist + xlist)
+        
+        added_timeout = timeout
```

Comment:
I prefer one-liner ifs in that case: `added_timeout = float('inf') if timeout is None else timeout`. Only if it's below 80 cols though :^)